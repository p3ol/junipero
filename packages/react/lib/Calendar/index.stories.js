import Calendar from './';

export default { title: 'react/Calendar' };

export const basic = () => <Calendar />;
