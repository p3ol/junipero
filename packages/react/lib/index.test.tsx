import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import type { FieldContent } from './types';
import TextField from './TextField';

interface State {
  value: {
    text: string;
    content: string[];
  };
}

describe('mockState: inside react', () => {
  it('should correctly update the state and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState<State>, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }: FieldContent) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
