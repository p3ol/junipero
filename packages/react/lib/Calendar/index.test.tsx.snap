// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Calendar /> > should allow to navigate between months > next month 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          January
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info active"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        10
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should allow to navigate between months > previous month 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          December
        </strong>
         2018
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        6
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should have a 3 days limited datepicker when min/max is set 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          December
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info active"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        5
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should not let you pick a disabled date 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          December
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive disabled"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info disabled"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info active"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should render > default 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          February
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info active"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        10
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should switch to next month when right arrow is clicked even if the 31st day is selected 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          February
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        10
      </a>
    </div>
  </div>
</div>
`;

exports[`<Calendar /> > should switch to previous month when left arrow is clicked even if the 31th day is selected 1`] = `
<div>
  <div
    class="junipero calendar"
  >
    <div
      class="calendar-header"
    >
      <div
        class="current-month"
      >
        <strong>
          November
        </strong>
         2019
      </div>
      <div
        class="navigation"
      >
        <svg
          class="junipero icon arrow-left"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M5.59863 9.04065L1.66113 5.04065L5.59863 1.04065"
          />
        </svg>
        <svg
          class="junipero icon arrow-right"
          height="11"
          viewBox="0 0 7 11"
          width="7"
        >
          <path
            d="M1.72388 9.04065L5.66138 5.04065L1.72388 1.04065"
          />
        </svg>
      </div>
    </div>
    <div
      class="calendar-body"
    >
      <span
        class="week-day junipero info"
      >
        Mon
      </span>
      <span
        class="week-day junipero info"
      >
        Tue
      </span>
      <span
        class="week-day junipero info"
      >
        Wen
      </span>
      <span
        class="week-day junipero info"
      >
        Thu
      </span>
      <span
        class="week-day junipero info"
      >
        Fri
      </span>
      <span
        class="week-day junipero info"
      >
        Sat
      </span>
      <span
        class="week-day junipero info"
      >
        Sun
      </span>
      <a
        class="day junipero info inactive"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        31
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        8
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        9
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        10
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        11
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        12
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        13
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        14
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        15
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        16
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        17
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        18
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        19
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        20
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        21
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        22
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        23
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        24
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        25
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        26
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        27
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        28
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        29
      </a>
      <a
        class="day junipero info"
        href="#"
      >
        30
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        1
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        2
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        3
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        4
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        5
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        6
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        7
      </a>
      <a
        class="day junipero info inactive"
        href="#"
      >
        8
      </a>
    </div>
  </div>
</div>
`;
